tools/ProductConstructionService.ReproTool/Operations/FlowCommitOperation.cs

```diff
@@ -149,4 +151,40 @@ internal override async Task RunAsync()
 
         _logger.LogInformation("Subscription triggered. Wait for a PR in {url}", $"{_options.TargetRepository}/pulls");
     }
+
+    private async Task<bool> IsRepoVmr(string repoName, string repoOwner)
+    {
+        try
+        {
+            await _ghClient.Repository.Content.GetAllContents(repoOwner, repoName, SourceManifestPath);
+            return true;
+        }
+        catch (Octokit.ApiException e)
+        {
+            if (e.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return false;
+            }
+            throw;
+        }
+    }
+
+    private async Task<(bool isSourceEnabled, bool isForwardFlow)> GetCodeflowMetadata(
+        string sourceUri,
+        string targetUri)
+    {
+        var (targetRepo, targetOwner) = GitRepoUrlUtils.GetRepoNameAndOwner(targetUri);
```

Comment:
nit: we could pass these as method arguments, since we calculate them right before calling this method